pyccel/parser/scope.py

```diff
@@ -0,0 +1,319 @@
+            New child scope, which has the current object as parent.
+
+        """
+
+        child = Scope(**kwargs, parent_scope = self)
+
+        self._sons_scopes[name] = child
+        child.parent_scope = self
+
+        return child
+
+    @property
+    def imports(self):
+        """ A dictionary of objects imported in this scope
+        """
+        return self._imports
+
+    @property
+    def variables(self):
+        """ A dictionary of variables defined in this scope
+        """
+        return self._locals['variables']
+
+    @property
+    def classes(self):
+        """ A dictionary of classes defined in this scope
+        """
+        return self._locals['classes']
+
+    @property
+    def functions(self):
+        """ A dictionary of functions defined in this scope
+        """
+        return self._locals['functions']
+
+    @property
+    def macros(self):
+        """ A dictionary of macros defined in this scope
+        """
+        return self._locals['macros']
+
+    @property
+    def headers(self):
+        """A dictionary of user defined headers which may
+        be applied to functions in this scope"""
+        return self._locals['headers']
+
+    @property
+    def templates(self):
+        """A dictionary of user defined templates which may
+        be applied to functions in this scope"""
+        return self._locals['templates']
+
+    @property
+    def decorators(self):
+        """Dictionary of Pyccel decorators which may be
+        applied to a function definition in this scope."""
+        return self._locals['decorators']
+
+    @property
+    def cls_constructs(self):
+        """ A dictionary of datatypes for the classes defined in
+        this scope
+        """
+        return self._locals['cls_constructs']
+
+    @property
+    def sons_scopes(self):
+        """ A dictionary of all the scopes contained within the
+        current scope
+        """
+        return self._sons_scopes
+
+    @property
+    def symbolic_functions(self):
+        """ A dictionary of symbolic functions defined in this scope
+        """
+        return self._locals['symbolic_functions']
+
+    def find_in_scope(self, name, location = None):
+        """ Find and return the specified object in the scope
+        If the object cannot be found then None is returned
+        """
+        for l in ([location] if location else self._locals.keys()):
+            if name in self._locals[l]:
+                return self._locals[l][name]
+
+            if name in self.imports[l]:
+                return self.imports[l][name]
+
+        # Walk up the tree of Scope objects, until the root if needed
+        if self.parent_scope:
+            return self.parent_scope.find_in_scope(name, location)
+        else:
+            return None
+
+    def get_all_from_scope(self, location):
+        """ Find and return the specified object in the scope
+        If the object cannot be found then None is returned
+        """
```

Comment:
Here I would simply change the method name from `get_all_from_scope` to `find_all`

Reply:
Fixed [c4ead6c](https://github.com/pyccel/pyccel/pull/1068/commits/c4ead6c1765b206ee9e13c6a5a787277321f4e75)